pkg/retry/retry.go

```diff
@@ -93,18 +93,19 @@ func (s *Server) handleRetry() http.Handler {
 
 		logger.Infow("retrieved last checkpoint", "lastCheckpoint", lastCheckpoint)
 
-		// for each run track the total number of events retrieve and failure events
 		var totalEventCount int
 		var failedEventCount int
+		var redeliveredEventCount int
 		var newCheckpoint string
 		var cursor string
 
 		// store all observed failures in memory from the latest event up to the lastCheckpoint
 		var failedEventsHistory []*eventIdentifier
+		var found bool
 
 		// the first run of this service will not have a cursor therefore we must
 		// ensure we run the loop at least once
-		for ok := true; ok; ok = (cursor != "") {
+		for ok := true; ok; ok = (cursor != "" && !found) {
```

Comment:
currently we are exhaustively calling for all deliveries, this should stop that unintended behavior